libs/MeshKernel/src/CurvilinearGrid/CurvilinearGrid.cpp

```diff
@@ -80,6 +80,62 @@ CurvilinearGrid::CurvilinearGrid(lin_alg::Matrix<Point>&& grid, Projection proje
     SetGridNodes(std::move(grid));
 }
 
+CurvilinearGrid& CurvilinearGrid::operator=(CurvilinearGrid&& copy) noexcept
+{
+    if (this != &copy)
+    {
+        m_projection = copy.m_projection;
+        m_gridNodes = std::move(copy.m_gridNodes);
+        m_gridFacesMask = std::move(copy.m_gridFacesMask);
+        m_gridNodesTypes = std::move(copy.m_gridNodesTypes);
+        m_gridIndices = std::move(copy.m_gridIndices);
+
+        m_nodesRTreeRequiresUpdate = copy.m_nodesRTreeRequiresUpdate;
+        m_edgesRTreeRequiresUpdate = copy.m_edgesRTreeRequiresUpdate;
+        m_facesRTreeRequiresUpdate = copy.m_facesRTreeRequiresUpdate;
+        m_RTrees = std::move(copy.m_RTrees);
+        m_boundingBoxCache = copy.m_boundingBoxCache;
+
+        m_edges = std::move(copy.m_edges);
+
+        m_startOffset = copy.m_startOffset;
```

Comment:
Set the moved-from object to a default valid state? i think is required for moved objects

        copy.m_projection = nullptr;  // Assuming m_projection is a pointer, otherwise initialize it appropriately
        copy.m_nodesRTreeRequiresUpdate = false;
        copy.m_edgesRTreeRequiresUpdate = false;
        copy.m_facesRTreeRequiresUpdate = false;
        copy.m_startOffset = 0;
        copy.m_endOffset = 0;

Reply:
done